docs/design/ttl_scans.md

```diff
@@ -0,0 +1,78 @@
+# TTL scans
+
+## Summary
+
+Add an option to automatically delete old starboard scans, in this first version focus on vulnerability scans but in the long run we could add similar functionality to other reports as well.
+
+## Motivation
+
+In [537](https://github.com/aquasecurity/starboard/issues/537) we talk about a need to run nightly vulnerability scans of CVE:s.
+This way we can make sure to get new CVE reports for long time running pods as well.
+
+## Proposal
+
+Add a environment variable to the operator, for example `OPERATOR_VULNERABILITY_SCANNER_REPORT_TTL=86400`, this way we can add other ttl values for other reports as well.
+Adding this environment variable would add a annotation to the generated VulnerabilityReport that we can look for.
+
+Create a new controller that looks for changes in vulnerabilityreports and uses [RequeueAfter](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Result)
+to look through all existing vulnerabilityreports at startup of the operator but also monitor newly created reports and requeue them after need
```

Comment:
I think that with RequeAfter property of reconciliation result we could avoid listing vulnerability reports at startup at all. The goal is to avoid batch processing as much as we can (especially on startup).

Therefore, I'd clarify that the annotation and the update timestamp is used to determine a TTL for a single report. If the report should be kept we adding it back to the work queue. Otherwise it's deleted.

Reply:
Rewrote the section and I hope It made it clearer.